pkg/catalog/types.go

```diff
@@ -0,0 +1,48 @@
+package catalog
+
+import (
+	imagev1 "github.com/openshift/api/image/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ComponentType ...
+type ComponentType struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ComponentSpec `json:"spec,omitempty"`
+}
+
+// ComponentSpec ...
+type ComponentSpec struct {
+	Namespace      string              `json:"namespace"`
```

Comment:
Namespace filed is already in  `metav1.ObjectMeta` (metadata), we should use that one.